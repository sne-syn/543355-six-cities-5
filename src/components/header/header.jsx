import React from 'react';
import {Link} from 'react-router-dom';
import PropTypes from 'prop-types';
import {AppRoute} from "../../utils/const";

const Header = ({isLogged}) => {
  return (
    <header className="header">
      <div className="container">
        <div className="header__wrapper">
          <div className="header__left">
            <Link to={AppRoute.ROOT} className="header__logo-link header__logo-link--active">
              <img className="header__logo" src="/img/logo.svg" alt="6 cities logo" width="81" height="41" />
            </Link>
          </div>
          <nav className="header__nav">
            <ul className="header__nav-list">
              <li className="header__nav-item user">
                {isLogged ? (<Link to={AppRoute.FAVORITES} className="header__nav-link header__nav-link--profile" href="#">
                  <div className="header__avatar-wrapper user__avatar-wrapper">
                  </div>
                  <span className="header__user-name user__name">Oliver.conner@gmail.com</span>
                </Link>) : (<Link to={AppRoute.LOGIN} className="header__nav-link header__nav-link--profile" href="#">
                  <div className="header__avatar-wrapper user__avatar-wrapper">
                  </div><span className="header__login">Sign in</span></Link>)}
              </li>
            </ul>
          </nav>
        </div>
      </div>
    </header>
  );
};

Header.propTypes = {
  isLogged: PropTypes.bool.isRequired,
};

export default React.memo(Header);
