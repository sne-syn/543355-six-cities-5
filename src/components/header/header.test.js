import React from 'react';
import renderer from 'react-test-renderer';
import {Header} from './header';

jest.mock(`react-router-dom`, () => ({Link: `Link`}));

test(`Render Header`, () => {
  const tree = renderer
    .create(<Header userEmail={``} authorizationStatus={`NO_AUTH`} />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
