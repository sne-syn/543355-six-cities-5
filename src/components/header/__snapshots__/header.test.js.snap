// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Header 1`] = `
<header
  className="header"
>
  <div
    className="container"
  >
    <div
      className="header__wrapper"
    >
      <div
        className="header__left"
      >
        <Link
          className="header__logo-link header__logo-link--active"
          to="/"
        >
          <img
            alt="6 cities logo"
            className="header__logo"
            height="41"
            src="/img/logo.svg"
            width="81"
          />
        </Link>
      </div>
      <nav
        className="header__nav"
      >
        <ul
          className="header__nav-list"
        >
          <li
            className="header__nav-item user"
          >
            <Link
              className="header__nav-link header__nav-link--profile"
              href="#"
              to="/login"
            >
              <div
                className="header__avatar-wrapper user__avatar-wrapper"
              />
              <span
                className="header__login"
              >
                Sign in
              </span>
            </Link>
          </li>
        </ul>
      </nav>
    </div>
  </div>
</header>
`;
